internal/controllers/clustercatalog_controller.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
+)
+
+type CatalogCacheRemover interface {
+	Remove(catalogName string) error
+}
+
+// ClusterCatalogReconciler reconciles a ClusterCatalog object
+type ClusterCatalogReconciler struct {
+	client.Client
+	Cache CatalogCacheRemover
+}
+
+//+kubebuilder:rbac:groups=olm.operatorframework.io,resources=clustercatalogs,verbs=get;list;watch
+
+func (r *ClusterCatalogReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	existingCatalog := &catalogd.ClusterCatalog{}
+	err := r.Client.Get(ctx, req.NamespacedName, existingCatalog)
+	if client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, err
+	}
+	if apierrors.IsNotFound(err) {
+		return ctrl.Result{}, r.Cache.Remove(req.Name)
```

Comment:
Should we create one or more new metrics related to this cache? For example:

- gauge: number of known ClusterCatalogs
- gauge: number of local caches

Or maybe just store the diff as a single gauge (that we would expect to always be 0)?

Reply:
I don't think this is unreasonable, but I would probably consider this outside the scope of this specific PR. I _do_ think we should eventually have a discussion about instrumenting OLMv1 with meaningful metrics.

Reply:
Fair!

Reply:
I've created #1287 to keep track of this. Are there any labels we'd want to apply to the issue?